app/src/main/java/net/gsantner/opoc/frontend/base/GsActivityBase.java

```diff
@@ -83,6 +87,21 @@ protected void onResume() {
         m_setActivityNavigationBarColor.callback();
     }
 
+    @Override
+    protected void onPostResume() {
+        super.onPostResume();
+        // Only do this when not being restored
+        if (_activityFirstTimeVisible.getAndSet(false) && _savedInstanceState == null) {
+            new Handler(getMainLooper()).post(this::onActivityFirstTimeVisible);
+        }
+    }
+
+    /**
+     * This will be called when this activity gets the first time visible
+     */
+    public void onActivityFirstTimeVisible() {
```

Comment:
Added to activities too.

Reply:
Makes sense, unfortunately hard to make something that works in the same way at fragment and activity, and to share code between. Unfortunately similar but not same 😄 .